app/models/manageiq/providers/vmware/cloud_manager/vm/operations.rb

```diff
@@ -1,3 +1,23 @@
 module ManageIQ::Providers::Vmware::CloudManager::Vm::Operations
+  extend ActiveSupport::Concern
+
   include_concern 'Power'
+
+  def raw_destroy
+    raise "VM has no #{ui_lookup(:table => "ext_management_systems")}, unable to destroy VM" unless ext_management_system
+    ext_management_system.with_provider_connection do |service|
+      begin
+        response = service.post_undeploy_vapp(ems_ref, :UndeployPowerAction => 'shutdown')
+        service.process_task(response.body)
+      rescue => err
+        $vcloud_log.debug("vm=[#{name}], error: VM not running! Ignoring error: #{err}")
+      ensure
+        response = service.delete_vapp(ems_ref)
```

Comment:
Don't like doing this in the ensure block assuming that you're rescuing a "vm already powered off" exception because it could be something completely different and we'll still try to do this.
I'd rather rescue the exact exception then do this outside the begin/rescue block like:

```
begin
  service.post_undeploy_vapp(ems_ref, :UndeployPowerAction => 'shutdown')
rescue Vcloud::VmAlreadyPoweredOff /* not sure what this exception actually is, just for demonstration */
  # this is expected if the vapp is already off
end

service.process_task(response.body)
update_attributes!(:raw_power_state => "terminated")
```